django/contrib/auth/forms.py

```diff
@@ -378,12 +378,11 @@ def __init__(self, user, *args, **kwargs):
     def clean_new_password2(self):
         password1 = self.cleaned_data.get("new_password1")
         password2 = self.cleaned_data.get("new_password2")
-        if password1 and password2:
-            if password1 != password2:
-                raise ValidationError(
-                    self.error_messages["password_mismatch"],
-                    code="password_mismatch",
-                )
+        if password1 and password2 and password1 != password2:
```

Comment:
password1 and password2 are required form fields, so I'm wondering whether the empty/None check here is superfluous? 🤔

Reply:
Hi shangxiao! Thanks for the review.

I thought the same!

But when I ran the tests I realized that we need the two variables to raise the "didn't match error".

<img width="1048" alt="image" src="https://user-images.githubusercontent.com/16822952/197392787-48ff9820-b9b9-4d5e-ab86-30c3144fef72.png">

`UserCreationForm`, `SetPasswordForm` and now `AdminPasswordChangeForm` have the same condition!